src/router_core/modules/edge_router/addr_proxy.c

```diff
@@ -215,6 +215,50 @@ static void del_outlink(qcm_edge_addr_proxy_t *ap, qdr_address_t *addr)
     }
 }
 
+static void on_link_event(void *context, qdrc_event_t event, qdr_link_t *link)
+{
+    if (!link || !link->conn)
+        return;
+
+    //
+    // We only care if the link event is on an edge connection.
+    //
+    if (link->conn->role != QDR_ROLE_EDGE_CONNECTION)
+            return;
+
+    switch (event) {
+        case QDRC_EVENT_LINK_OUT_DETACHED: {
+            qdr_address_t *addr = link->owning_addr;
+            if (addr && addr->edge_outlink && link == addr->edge_outlink) {
```

Comment:
The second condition in the if is technically not necessary.  edge_outlink is never dereferenced, only compared.